vizro-ai/src/vizro_ai/_vizro_ai.py

```diff
@@ -64,13 +63,17 @@ def _run_plot_tasks(
         self, df: pd.DataFrame, user_input: str, max_debug_retry: int = 3, explain: bool = False
     ) -> Dict[str, Any]:
         """Task execution."""
-        target_chart = self._lazy_get_component(GetChartSelection).run(df=df, chain_input=user_input)
-        df_code = self._lazy_get_component(GetDataFrameCraft).run(df=df, chain_input=user_input)
-        visual_code = self._lazy_get_component(GetVisualCode).run(
-            chain_input=user_input, chart_types=target_chart, df_code=df_code
+        self.pipeline_manager.llm = self.llm_to_use
```

Comment:
I am not 100%, but since we define `PipelineManager` as a class attribute, would this mean that setting the LLM would alter it for all instances of VizroAI as well? If yes, is that our desired behaviour?

Reply:
yes i think we want to use the same model for the all of the LLM calls in each components / task pipelines once it is set in initialization of VizroAI? 

but probably it is better to fix this pipeline manager with the LLM outside of this plot tasks,  better in initialization of `VizroAI` if we have multiple tasks in future, adding the change now.

Reply:
What I meant was, do we want the following case:

```py
vizro_ai_A = VizroAI(model="foo")

vizro_ai_B = VizroAI(model="bar")

# Then
vizro_AI_A.plot(df,"...")
# would use model="bar" no
```

As a class is a mutable object, and hence we would change it for all classes? Or am I missing something?

Reply:
i don't think so, objects including instance is mutable just through its own. If you have two instances of a class, mutating one does not inherently mutate the other. They have their own separate attributes and states.
```
class PipelineManager:
    def __int__(self,llm):
        self.llm = llm

class Test:
    _pipeline_manager = PipelineManager()
    def __init__(self, llm):
        self.llm = llm
        self._pipeline_manager.llm = llm

    def print_llm(self):
        print(self.llm)

test_a = Test("A")
test_b = Test("B")

test_a.print_llm()
test_b.print_llm()
# A
# B
```